### DISCUSSION TITLE
no applicable stereotypes

### DISCUSSION THREAD

**Discussion by erichstuder** [eclipse-acceleo/acceleo#155] [Q&A]
Hello

I have to following template to play around with.
It finds the classes in my model an parses them.
The only thing is that getApplicableStereotypes() returns zero.
How can that be?
I expected if a class is found there should be some applicable stereotypes.
What am I getting wrong?

```
[template public generateElement(element : Element)]
[comment @main /]
[if element.oclIsTypeOf(Class)]
    [file ('class_' + element.oclAsType(uml::Class).name, false, 'UTF-8')]
        [let appliedStereotypes : Set(uml::Stereotype) = element.oclAsType(uml::Class).getApplicableStereotypes()]
            [appliedStereotypes->size()/] stereotypes found:
            [for (stereotype : uml::Stereotype | appliedStereotypes)]
                [stereotype.name + ' aaa'/]
            [/for]
        [/let]
    [/file]
[/if]
[/template]
```
[/template]

**Comment by ylussaud** [comment]
I'm not an UML expert but I think the Package containing the Class should have a profile applied so UML can compute available Stereotype for your Class. You can see this [thread](https://www.eclipse.org/forums/index.php/t/1096252/) from the Eclipse forum.

Also you might need to add the initialization of the ResourceSet in the main Class generated by Acceleo:
```
org.eclipse.uml2.uml.resources.util.UMLResourcesUtil.init(ResourceSet)
```

Don't forget to add `@generated NOT` in the comment above the method.

**Reply by erichstuder** [reply]
Thank your for the answer @ylussaud 

The solution was the code snippet you suggested initializing ResourceSet.
In the Code there is a comment, that this is only necessary for standalone setups.
As I have everything in Eclipse I would never have thought of doing this.

It looks like working for uml elements now.
But i have a sysml 1.6 Diagram and i read that there is no such function.
But I don't know how up to date these texts were.
Is there any hope to achieve the same with a sysml diagram?

Acceleo really looks like a simple but powerful tool.
There are some tutorials, but never know how up to date they are and also they mostly speak of very simple cases.
I'm not very experienced in the Eclipse-World so it would be nice to have a documentation that also explains a bit how it works behind the scene. This might make it easier for me to figure out problems without browsing the code. But I might be wrong.

Thank you
erichstuder

**Comment by ylussaud** [comment]
We a in the process of writing a new version of Acceleo (Acceleo 4) and yes a lot of resources around Acceleo 3.x are old. The Java main is ran as a Java application outside of Eclipse. I'm not a SysML expert but there is this [issue](https://bugs.eclipse.org/bugs/show_bug.cgi?id=531840) explaining why the SysML ResourceSet initialization has been removed. I guess it is no longer needed.
